components/announcement_bar/cloud_trial_announcement_bar/index.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {getStandardAnalytics} from 'mattermost-redux/actions/admin';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import {getCloudSubscription} from 'mattermost-redux/actions/cloud';
+
+import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+
+import {openModal} from 'actions/views/modals';
+
+import {GlobalState} from 'types/store';
+
+import {Preferences} from 'utils/constants';
+
+import {getRemainingDaysFromFutureTimestamp} from 'utils/utils.jsx';
+
+import CloudTrialAnnouncementBar from './cloud_trial_announcement_bar';
+
+function mapStateToProps(state: GlobalState) {
+    const getCategory = makeGetCategory();
+
+    const subscription = state.entities.cloud.subscription;
+    let isFreeTrial = false;
+    let daysLeftOnTrial = 0;
+
+    if (subscription!.end_at > 0 && subscription!.status === 'trialing') {
```

Comment:
I think it should be safer to do a null check here instead of making the assertion that it will not be null, if the user is not admin they do not have permissions on the subscription I believe

```suggestion
    if (subscription?.end_at > 0 && subscription?.status === 'trialing') {
```